### PR TITLE
replace QMessageBox with QInputDialog ensuring that the user typed th…

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
…e full name of the project to delete

__Confirmation prompt (first time)__

image

__Confirmation prompt (retry__

Capture d’écran du 2023-05-30 15-27-14

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-2571 Add "username/project_name" confirmation to delete project button in QFieldSync](https://app.clickup.com/t/2192114/QF-2571)

File: qfieldsync/gui/cloud_projects_dialog.py [hunk 1]
```diff
@@ -790,21 +791,18 @@ def on_project_launch_button_clicked(self) -> None:
...
 
     def on_project_delete_button_clicked(self) -> None:
-        button_pressed = QMessageBox.question(
-            self,
+        text, ok = QInputDialog().getText(
+            self, 
             self.tr("Delete QFieldCloud project"),
             self.tr(
-                'Are you sure you want to delete the QFieldCloud project "{}"? Nevertheless, your local files will remain.'
+                'Are you sure you want to delete your QFieldCloud project? If so please type "{}". NB: Your local files will not be deleted.'
```
Comment (m-kuhn) [inline qfieldsync/gui/cloud_projects_dialog.py:798]:
```suggestion
                'Are you sure you want to delete your QFieldCloud project? If so please type <b>{}</b>. NB: Your local files will not be deleted.'
```

File: qfieldsync/gui/cloud_projects_dialog.py [hunk 2]
```diff
@@ -790,21 +791,18 @@ def on_project_launch_button_clicked(self) -> None:
...
         )
 
-        if button_pressed != QMessageBox.Yes:
-            return
-
-        self.projectsStack.setEnabled(False)
-
-        reply = self.network_manager.delete_project(self.current_cloud_project.id)
-        reply.finished.connect(lambda: self.on_delete_project_reply_finished(reply))
+        if ok and text == self.current_cloud_project.name:
```
Comment (m-kuhn) [inline qfieldsync/gui/cloud_projects_dialog.py:802]:
```suggestion
        if ok and text.strip() == self.current_cloud_project.name:
```

File: qfieldsync/gui/cloud_projects_dialog.py [hunk 3]
```diff
@@ -791,19 +791,28 @@ def on_project_launch_button_clicked(self) -> None:
...
     def on_project_delete_button_clicked(self) -> None:
-        text, ok = QInputDialog().getText(
-            self, 
-            self.tr("Delete QFieldCloud project"),
-            self.tr(
-                'Are you sure you want to delete your QFieldCloud project? If so please type "{}". NB: Your local files will not be deleted.'
-            ).format(self.current_cloud_project.name),
-        )
+        def ask(remark: Optional[str] = None):
+            remark = remark or "Are you sure you want to delete this QFieldCloud project? If so "
```
Comment (m-kuhn) [inline qfieldsync/gui/cloud_projects_dialog.py:795]: Watch out for translation here

**Comment by why-not-try-calmer** [comment]
Hum, I don't really see how failures in the test related to the PR. Is the test suite functional?

**Comment by m-kuhn** [comment]
> Hum, I don't really see how the failures in the tests are related to the PR. Is the test suite functional?

The `qgis/qgis:latest` image is broken

**Comment by m-kuhn** [comment]
Thanks for the screenshot.

- Can you make sure it's `username/project_name` which is used as confirmation (there might be identically named projects under your personal account and an org)?
- I assume it's already the first time bold?
- I would also leave the "Are you sure..." message the second time. It adds context.

File: qfieldsync/gui/cloud_projects_dialog.py [hunk 4]
```diff
@@ -790,21 +791,34 @@ def on_project_launch_button_clicked(self) -> None:
...
-        if button_pressed != QMessageBox.Yes:
-            return
+        text, ok = ask()
 
-        self.projectsStack.setEnabled(False)
+        if ok:
+            clean_text = text.strip()
+            while clean_text != self.current_cloud_project.name:
+                updated_text, updated_ok = ask(
+                    "<p style='color:red'>Incorrect project name!</p>"
```
Comment (m-kuhn) [inline qfieldsync/gui/cloud_projects_dialog.py:813]: Can you make sure everything is translatable?

File: qfieldsync/gui/cloud_projects_dialog.py [hunk 5]
```diff
@@ -801,14 +805,14 @@ def ask(remark: Optional[str] = None):
                 self.tr("Delete QFieldCloud project"),
                 self.tr(
                     '{} To confirm deletion please type "<em>{}</em>". NB: Your local files will not be deleted.'
```
[the remark this answers is not in the document]
Comment (m-kuhn) [reply qfieldsync/gui/cloud_projects_dialog.py]: @why-not-try-calmer bump

[the remark this answers is not in the document]
Comment (why-not-try-calmer) [reply qfieldsync/gui/cloud_projects_dialog.py]: > Are we sure the local copy stays unaffected?
Insofar as `self.network_manager.delete_project(self.current_cloud_project.id)` resolves to a call to `QNetworkAccessManager::deleteResource`, I would say Yes.

**Comment by why-not-try-calmer** [comment]
Updated screenshots:

__Confirmation prompt (first time)__
Capture d’écran du 2023-05-30 21-40-17

__Confirmation prompt (retry)__
Capture d’écran du 2023-05-30 21-40-25

**Comment by m-kuhn** [comment]
> > Hum, I don't really see how the failures in the tests are related to the PR. Is the test suite functional?
> 
> The `qgis/qgis:latest` image is broken

https://github.com/qgis/QGIS/pull/53328

**Review by m-kuhn — APPROVED** [review]
Thank you, good to merge.

No need to wait for the fix for master

**Pull request merged**